delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala

```diff
@@ -45,13 +47,15 @@ object SearchParams {
     * @param rev        the optional revision of the realm resources
     * @param createdBy  the optional subject who created the realm resource
     * @param updatedBy  the optional subject who updated the realm resource
+    * @param filter     the additional filter to select realms
     */
   final case class RealmSearchParams(
       issuer: Option[String] = None,
       deprecated: Option[Boolean] = None,
       rev: Option[Long] = None,
       createdBy: Option[Subject] = None,
-      updatedBy: Option[Subject] = None
+      updatedBy: Option[Subject] = None,
+      filter: Realm => Boolean = _ => true
```

Comment:
I think it is better not put a default value on this param as it is the most critical one

Reply:
It's more consistent with other params which are all optional and you can override it only when you need it like the other params.

Reply:
The default value here can give you access to resources you shouldn't have access to and it can be easy to misuse it silently.
That is why I wouldn't put a default value like the others because if we screw up for the others, the result is false but doesn't expose data you should not have access to.

Reply:
I guess it would make sense to not have the default for `OrganizationSearchParams` and `ProjectSearchParams`, but for the `RealmSearchParams` we'll probably never going to use the filter, so it make sense to have a default there.

Reply:
In this case we probably also need to remove the default value for organizationParams in this function
```scala
 def list(
      pagination: FromPagination,
      params: OrganizationSearchParams = OrganizationSearchParams.none
  ): UIO[UnscoredSearchResults[OrganizationResource]]
```
 in `Organizations` since there is no logical default anymore.
And similarly in `Projects`.